### PR TITLE
Verification: elegant methods for callouts

### PR DESCRIPTION
**Pull request by 650elx**

### CODE REVIEW

File: sinch/domains/verification/models/responses.py [hunk 1]
```diff
@@ -12,22 +31,22 @@ class StartVerificationResponse(SinchBaseModel):
...
 @dataclass
 class StartSMSInitiateVerificationResponse(StartVerificationResponse):
-    sms: dict
+    sms: SMSResponse
 
 
 @dataclass
 class StartFlashCallInitiateVerificationResponse(StartVerificationResponse):
-    flashcall: dict
+    flashcall: FlashCallResponse
```
Comment (asein-sinch) [inline sinch/domains/verification/models/responses.py:39]: The server returns a property named `flashCall`. Is there a reason to change it to `flashcall`?

File: sinch/domains/verification/endpoints/start_verification.py [hunk 2]
```diff
@@ -32,20 +35,38 @@ def handle_response(self, response: HTTPResponse) -> StartVerificationResponse:
         super().handle_response(response)
         if self.request_data.method == VerificationMethod.SMS.value:
             return StartSMSInitiateVerificationResponse(
-                **response.body
+                id=response.body.get("id"),
+                method=response.body.get("method"),
+                _links=response.body.get("_links"),
+                sms=SMSResponse(
+                    interception_timeout=response.body["sms"].get("interceptionTimeout"),
```
Comment (asein-sinch) [inline sinch/domains/verification/endpoints/start_verification.py:42]: We can expect the server to always return this field, but it's noted as optional in the specification. Wouldn't it be safer to check for the existence of the `sms` property before reading sub-properties?

File: sinch/domains/verification/models/__init__.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+from typing import TypedDict
+
+
+class VerificationIdentity(TypedDict):
+    type: str
```
Comment (asein-sinch) [inline sinch/domains/verification/models/__init__.py:5]: The spec mentions that only `number` is accepted as a value for this property. Maybe you can give a hint to the user about valid values

**Review by asein-sinch — DISMISSED** [review]

File: sinch/domains/verification/models/responses.py (hunk 1 above)
Comment (650elx) [reply sinch/domains/verification/models/responses.py:39]: I want to be consistent and return values in `snake_case` not `camelCase`.
`flash_call` vs `flashcall` ? What's better? Wdyt?

File: sinch/domains/verification/models/__init__.py (hunk 3 above)
Comment (650elx) [reply sinch/domains/verification/models/__init__.py]: Why not. Literal type hint added.

File: sinch/domains/verification/models/responses.py (hunk 1 above)
Comment (asein-sinch) [reply sinch/domains/verification/models/responses.py:39]: That's up to you 🙂

File: sinch/domains/verification/endpoints/start_verification.py (hunk 2 above)
Comment (650elx) [reply sinch/domains/verification/endpoints/start_verification.py:43]: I've added some additional logic.

**Review by asein-sinch — APPROVED** [review]

**Pull request merged**